### PR TITLE
Remove UiKit

### LINKED ISSUE #538: Remove uikit
- [ ] Replace uikit classes with own ones where needed. For example: "uk-width-small" => "oc-width-small"
- [ ] Redefine uikit styles for now in "oc-" namespace (which can be adjusted later by a designer)
- [ ] Remove usage of uikit across Phoenix and make sure we have replacement "oc-" classes to use there if needed

@DeepDiver1975 @LukasHirt

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description & transition path
This PR removes the [UiKit](https://github.com/uikit/uikit/) library which the ODS had been built upon, vastly speeding up built times and reducing bundle & dependency size.

The strategy was searching & replacing currently used classes, porting over necessary ones from UiKit until we pick a successor (in terms of, e.g., helper classes). It now needs carefully testing with the current users of the ownCloud Design System before we can merge & release it.

Please note that some of the headings of the styleguidist docs now also have changed due to the UiKit removal, I didn't bother fixing those since we may (or may not) switch to Storybook which would make those effort obsolete. We can easily fix them later if necessary, though.

## Related Issue
- Fixes #538

## Types of changes
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Technical debt

## Open tasks:
- [ ] First quick code review @paulcod3 @kulmann 
- [ ] Create alpha release from this branch @pascalwengerter 
- [ ] Test web, file-picker with alpha release @pascalwengerter 
- [ ] Make sure internal repos work as expected with alpha release @dschmidt 
- [ ] Push fixes to this PR, do a beta (...) release, repeat above steps
- [ ] Collect other breaking changes we could include in this release (e.g. form prop naming)

### CODE REVIEW

File: src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue [hunk 1]
```diff
@@ -135,9 +141,15 @@ export default {
...
     > li button,
     > li span,
-    > :nth-child(n + 2):not(.uk-first-column)::before {
+    > :nth-child(n + 2)::before {
       color: var(--oc-color-text-muted);
     }
+    > :nth-child(n + 2)::before {
+      content: "/";
+      display: inline-block;
+      margin: 0 20px 0 calc(20px - 4px);
```
Comment (kulmann) [inline src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue:150]: Can you please reduce the margins by at least 50%?

Comment (pascalwengerter) [reply src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue:150]: Sure thing, done!

File: src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue [hunk 2]
```diff
@@ -252,15 +266,19 @@ export default {
...
 <template>
+<section>
   <div>
     <oc-breadcrumb :items="items" class="oc-mb" />
+  </div>
+  <div>
     <oc-breadcrumb :items="items" variation="lead" />
     <oc-breadcrumb :items="items" class="oc-mt-l">
       <template v-slot:contextMenu>
         <p>I'm an example item</p>
```
Comment (kulmann) [inline src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue:277]: The example list item has a margin top and no other margins or paddings. Was that intended? For the context menu we implemented a `paddingSize` prop in the `OcDrop`. Would be nice to be able to set that for the breadcrumbs as well, if you agree that that could be useful.


Screenshot 2022-01-10 at 22 05 37

File: src/components/molecules/OcTextInput/OcTextInput.vue [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div>
```
Comment (kulmann) [inline src/components/molecules/OcTextInput/OcTextInput.vue:2]: The `clear` cross icon has a weird vertical offset, see screenshot.

Screenshot 2022-01-10 at 22 13 10

File: src/styles/theme/_variables.scss [hunk 4]
```diff
@@ -3,3 +3,16 @@ $navbar-nav-item-height: 60px;
...
 $sizes: ("xs": $oc-space-xsmall, "s": $oc-space-small, "m": $oc-space-medium, "l": $oc-space-large, "xl": $oc-space-xlarge, "xxl": $oc-space-xxlarge);
+
+// these are the ones the removal of ukit complains about
+// copy them over accordingly and decide whether to add
+// to theme or not
+
+$global-control-height: 15px;
+$global-border-width: 2px;
+
+$global-control-small-height: 30px;
```
Comment (kulmann) [inline src/styles/theme/_variables.scss:14]: Why is the small height double the large height?

File: src/components/atoms/OcDrop/OcDrop.vue [hunk 5]
```diff
@@ -206,6 +202,7 @@ export default {
 }
 
 .oc-drop {
+  box-shadow: 0 5px 15px rgba(0,0,0,.08);
```
Comment (kulmann) [inline src/components/atoms/OcDrop/OcDrop.vue:205]: You could extend `.oc-box-shadow-medium` here, right?

**Review by kulmann — CHANGES_REQUESTED** [review]

**Review by kulmann — CHANGES_REQUESTED** [review]
- font family in docs doesn't look good 😢 
- OcCheckbox had a border radius before, seems lost now. it looked good IMO.
- search has broken icon positioning (text overlays icon, see e.g. `Filter` example in `OcDrop`)

File: src/components/organisms/OcRecipient/OcRecipient.vue [hunk 6]
```diff
@@ -123,7 +123,7 @@ export default {
 
 <docs>
```
Comment (lookacat) [inline src/components/organisms/OcRecipient/OcRecipient.vue:124]: Font of the OcRecipient looks weird

Comment (lookacat) [reply src/components/organisms/OcRecipient/OcRecipient.vue:124]: Ok sorry not only OcRecipient, but some components seem to be not affected by this

File: src/components/atoms/OcSearchBar/OcSearchBar.vue [hunk 7]
```diff
@@ -5,8 +5,8 @@
     class="oc-search"
     :class="{ 'oc-search-small': small }"
   >
```
Comment (lookacat) [inline src/components/atoms/OcSearchBar/OcSearchBar.vue:7]: OcSearchbar (small) looks weird in the docs

**Review by lookacat — COMMENTED** [review]
Font in some components is weird, overall looks pretty good

**Comment by kulmann** [comment]
Had the same impression, that fonts are weird in some components.

File: src/components/atoms/OcRadio/OcRadio.vue [hunk 8]
```diff
@@ -121,16 +121,33 @@ export default {
...
+  background-repeat: no-repeat;
+
+  display: inline-block;
+  margin: 0;
+  overflow: hidden;
+
+  transition: 0.2s ease-in-out;
+  transition-property: background-color, border;
+  vertical-align: middle;
+  width: 16px;
```
[the remark this answers is not in the document]
Comment (pascalwengerter) [reply src/components/atoms/OcRadio/OcRadio.vue]: Done

File: src/components/atoms/OcSearchBar/OcSearchBar.vue [hunk 9]
```diff
@@ -285,16 +295,13 @@ export default {
 
   &-small {
     .oc-search-input {
-      @extend .uk-form-small;
-
-      // Overrride default padding of small from
-      // TODO: Remove `!important` after we get rid of UIkit
-      padding-right: var(--oc-space-medium) !important;
+      height: 30px;
```
[the remark this answers is not in the document]
Comment (pascalwengerter) [reply src/components/atoms/OcSearchBar/OcSearchBar.vue:298]: Currently already the case IMO, see https://owncloud.design/#/oC%20Components/OcSearchBar

File: src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue (hunk 2 above)
Comment (pascalwengerter) [reply src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue:277]: You mean a breadcrumb-contextmenu-paddingSize?

File: src/components/molecules/OcTextInput/OcTextInput.vue (hunk 3 above)
Comment (pascalwengerter) [reply src/components/molecules/OcTextInput/OcTextInput.vue:2]: Good find, thx

File: src/components/atoms/OcDrop/OcDrop.vue (hunk 5 above)
Comment (pascalwengerter) [reply src/components/atoms/OcDrop/OcDrop.vue:205]: Yeah true, I'll add the class directly to the DOM element

File: src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue (hunk 2 above)
Comment (kulmann) [reply src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue:277]: yep, exactly :-)

Comment (kulmann) [reply src/components/molecules/OcBreadcrumb/OcBreadcrumb.vue:277]: with the same default as for the context menu in the table

File: src/styles/theme/_variables.scss (hunk 4 above)
Comment (pascalwengerter) [reply src/styles/theme/_variables.scss:14]: Dev leftover and great catch, I just pasted any values in there (see comment above)

**Comment by pascalwengerter** [comment]
@kulmann @lookacat ready for re-review (focus on last commit)

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**